WordPress/WordPressTest/PostCoordinatorTests.swift

```diff
@@ -0,0 +1,39 @@
+import UIKit
+import XCTest
+import Nimble
+
+@testable import WordPress
+
+class PostCoordinatorTests: XCTestCase {
+
+    private let context = TestContextManager().newDerivedContext()
+
+    func testDoNotUploadAPostWithFailedMedia() {
+        let post = PostBuilder(context).with(image: "test.jpeg", status: .failed).build()
+        let postServiceMock = PostServiceMock()
+        let postCoordinator = PostCoordinator(mainService: postServiceMock, backgroundService: postServiceMock)
+
+        postCoordinator.save(post: post)
+
+        expect(post.remoteStatus).toEventually(equal(.failed))
+        expect(postServiceMock.didCallUploadPost).to(beFalse())
+    }
+
+    func testUploadAPostWithNoFailedMedia() {
+        let post = PostBuilder(context).with(image: "test.jpeg").build()
+        let postServiceMock = PostServiceMock()
+        let postCoordinator = PostCoordinator(mainService: postServiceMock, backgroundService: postServiceMock)
+
+        postCoordinator.save(post: post)
+
+        expect(postServiceMock.didCallUploadPost).to(beTrue())
```

Comment:
I've added a similar test to prevent the regression that I introduced from happening again.

My first thought was to do exacly what you described but, as you said, it doesn't worth it.